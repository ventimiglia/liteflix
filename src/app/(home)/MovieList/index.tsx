"use client";

import CustomDropdown from "@/components/Dropdown";
import { useState } from "react";
import MovieCard from "../MovieCard";
import { AnimatePresence } from "framer-motion";

type Props = {
  popularMovies: Movie[];
  myMovies: Movie[];
  className?: string;
};

export const enum CATEGORY {
  POPULAR = "POPULARES",
  MY_MOVIES = "MIS PELICULAS",
}

const MovieList = ({ popularMovies, myMovies, className }: Props) => {
  const [selectedCategory, setSelectedCategory] = useState(CATEGORY.POPULAR);
  const movies =
    selectedCategory === CATEGORY.POPULAR ? popularMovies : myMovies;
  return (
    <aside
      className={`lg:animate-fade-left lg:animate-once lg:animate-delay-1000 lg:animate-ease-linear lg:animate-normal lg:animate-fill-both flex-col z-0 items-center justify-start h-full self-center w-56 min-w-max py-6 gap-4 overflow-y-auto overflow-x-hidden lg:max-h-[46rem] ${className}`}
    >
      <CustomDropdown
        selectedCategory={selectedCategory}
        setSelectedCategory={setSelectedCategory}
      />
      <AnimatePresence>
        {movies.length === 0 ? (
          <p className="text-white text-center">No hay peliculas</p>
        ) : (
          <>
            {movies.map((movie) => (
              <MovieCard key={movie.id} movie={movie} selectedCategory={selectedCategory} />
            ))}
          </>
        )}
      </AnimatePresence>
    </aside>
  );
};

export default MovieList;
