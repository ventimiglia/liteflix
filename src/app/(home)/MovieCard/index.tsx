import Image from "next/image";
import { CATEGORY } from "../MovieList";

const MovieCard = ({
  movies,
  category,
}: {
  movies: Movie[];
  category: CATEGORY;
}) => (
  <>
    {movies?.length > 0 ? (
      movies.map((movie) => (
        <article
          key={movie.id}
          className="flex flex-col justify-between gap-4 w-80 xl:w-56 h-44 xl:h-36 relative bg-gradient-to-t from-gray-800 to-transparent rounded-sm p-3 xl:min-h-[9rem]"
        >
          <Image
            src={`${
              category === CATEGORY.POPULAR
                ? `https://image.tmdb.org/t/p/original${movie.backdrop_path}`
                : movie.backdrop_path
            }`}
            alt={movie?.title || "Movie title"}
            fill
            priority
            className="absolute -z-10 object-cover w-full h-full rounded-sm"
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 30vw, 360px"
          />
          <div className="w-10 h-10 border border-white rounded-full bg-transparent text-center self-center relative top-1/2 -translate-y-1/2">
            <Image
              src="/play.svg"
              alt="Reproducir"
              width={14}
              height={14}
              className="relative top-1/2 left-1/2 transform -translate-x-[35%] -translate-y-1/2"
            />
          </div>
          <h3 className="text-center w-full overflow-ellipsis whitespace-nowrap overflow-hidden">
            {movie?.title}
          </h3>
        </article>
      ))
    ) : (
      <p>No hay peliculas</p>
    )}
  </>
);

export default MovieCard;
