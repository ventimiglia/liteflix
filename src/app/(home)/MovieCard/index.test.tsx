import React from "react";
import { render } from "@testing-library/react";
import MovieCard from "./";
import { CATEGORY } from "../MovieList";

describe("MovieCard", () => {
  const movie = {
    id: 1,
    title: "Test Movie",
    backdrop_path: "test.jpg",
    vote_average: 7.5,
    release_date: 2022,
  };

  const selectedCategory = CATEGORY.POPULAR;

  it("renders the movie card with default state", () => {
    const { container, getByText, getByAltText } = render(
      <MovieCard movie={movie} selectedCategory={selectedCategory} />
    );

    expect(getByText(movie.title)).toBeInTheDocument();
    expect(getByAltText("Reproducir")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

})