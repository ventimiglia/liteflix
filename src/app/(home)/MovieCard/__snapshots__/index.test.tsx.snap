// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders the movie card with default state 1`] = `
<div>
  <article
    class="flex flex-col justify-between gap-4 w-80 lg:w-56 h-44 lg:h-36 relative rounded-md p-3 lg:min-h-[9rem] bg-gradient-card transition-all"
  >
    <img
      alt="Test Movie"
      class="absolute -z-10 object-cover w-full h-full rounded-md"
      data-nimg="fill"
      decoding="async"
      fetchpriority="high"
      sizes="(max-width: 768px) 100vw, (max-width: 1200px) 30vw, 360px"
      src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=3840&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Foriginaltest.jpg&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <div
      class="w-10 h-10 border border-white rounded-full bg-transparent self-center relative top-1/2 -translate-y-1/2"
    >
      <img
        alt="Reproducir"
        class="w-auto h-auto relative top-1/2 left-1/2 transform -translate-x-[35%] -translate-y-1/2"
        data-nimg="1"
        decoding="async"
        height="14"
        loading="lazy"
        src="/play.svg"
        style="color: transparent;"
        width="14"
      />
    </div>
    <h3
      class="text-center w-full overflow-ellipsis whitespace-nowrap overflow-hidden"
      style="opacity: 0; transform: translateY(100%) translateZ(0);"
    >
      Test Movie
    </h3>
  </article>
</div>
`;
